x-pack/plugins/security_solution/public/detections/pages/detection_engine/rules/details/execution_log_table/execution_log_search_bar.tsx

```diff
@@ -0,0 +1,86 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import React from 'react';
+import { EuiHealth, EuiSearchBar, EuiSearchBarProps, SearchFilterConfig } from '@elastic/eui';
+import { RuleExecutionStatus } from '../../../../../../../common/detection_engine/schemas/common';
+import { getStatusColor } from '../../../../../components/rules/rule_execution_status/utils';
+
+import * as i18n from '../translations';
+
+interface ExecutionLogTableSearchProps {
+  onSearch: (args: Parameters<NonNullable<EuiSearchBarProps['onChange']>>[0]) => void;
+}
+
+export const EXECUTION_LOG_SEARCH_SCHEMA = {
+  strict: true,
+  fields: {
+    'kibana.alert.rule.execution.status': {
+      type: 'string',
+    },
+    '@timestamp': {
+      type: 'string',
+    },
+    'event.duration': {
+      type: 'string',
+    },
+    message: {
+      type: 'string',
+    },
+    'kibana.alert.rule.execution.metrics.total_alerts': {
+      type: 'string',
+    },
+    'kibana.alert.rule.execution.metrics.total_hits': {
+      type: 'string',
+    },
+    'kibana.alert.rule.execution.metrics.execution_gap_duration_s': {
+      type: 'string',
+    },
+    'kibana.alert.rule.execution.metrics.total_indexing_duration_ms': {
+      type: 'string',
+    },
+    'kibana.alert.rule.execution.metrics.total_search_duration_ms': {
+      type: 'string',
+    },
+  },
+};
+
+const statuses = (Object.keys(RuleExecutionStatus) as Array<keyof typeof RuleExecutionStatus>).map(
+  (key) => key
+);
+
+const filters: SearchFilterConfig[] = [
+  {
+    type: 'field_value_selection',
+    field: 'kibana.alert.rule.execution.status',
+    name: 'Status',
+    multiSelect: 'or',
+    options: statuses.map((status) => ({
+      value: status,
+      view: <EuiHealth color={getStatusColor(status)}>{status}</EuiHealth>,
+    })),
+  },
+];
+
+export const ExecutionLogSearchBar = React.memo<ExecutionLogTableSearchProps>(({ onSearch }) => {
```

Comment:
Really wanted to expose suggestions for the above `schema` but in checking with the EUI folks it seems you must use `EuiSuggest` for auto-complete, but that component does not support `FilterGroups` out of the box, so would need to wire that up separately (along with query text override when applying filters). 

In testing I found an issue with the `validation` feature of [`EuiSearchBar`](https://elastic.github.io/eui/#/forms/search-bar) where it reports `>` `<` as invalid operators. Even with specifying type as `number` in the schema I'm still seeing this, so will need to debug further, or just abandon to use `EuiSuggest` with custom filters that way we get auto-complete (or expose column header style filters like what was brought up previously).


<p align="center">
  <img width="500" src="https://user-images.githubusercontent.com/2946766/151910632-cb397ca9-6467-4c59-a40c-3f10c6ec8b2b.png" />
</p>